build.gradle

```diff
@@ -23,12 +23,20 @@ buildScan {
 subprojects {
     apply plugin: 'java'
     apply plugin: 'maven'
+    apply plugin: 'checkstyle'
 
     repositories {
         mavenLocal()
         mavenCentral()
     }
 
+    checkstyle {
+        // Stay on checkstyle version 6 until Java version 7 support can be dropped
+        toolVersion = '6.19'
+        showViolations = true
+        configFile = file("${project.rootDir}/checkstyle.xml")
```

Comment:
:+1: Great idea,
Originally I was thinking of moving rule by rule going from warning to error.
Going file by file could work as well, and it sound like using this to encourage community contributors to participate in the process could work out better.
As long as we don't try to do everything all at once it should work out. :smile: